### DISCUSSION TITLE
Find matching nodes from Styles Editing dialog

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#1128] [Ideas]
My mindmaps are built organically. That tends to generate weeds. Weeds that are in the nodes themselves, and weeds that are in styles that seemed like a good idea at the time, but were never really used.

Finding out if a style is well used in my mindmaps currently feels like a tedious process. I have to choose Styles from the item type drop down list, and then find the relevant style. Then I can use the Quick Filter to see how many nodes are in that style. It works, but I if my suggestion can be implemented, I think it would be faster and smoother.

My thinking is that the Style Editing dialog could serve as a way of automatically executing the Quick Filter while staying open. This would let me go style by style and quickly eliminate the weeds.

Another use case just occurred to me: I could arrange the styles such so as to quickly go through nodes in a certain sequences. For example, I could have nodes called "to do," "in progress" and "finished." Right next to each other. Double clicking the "to do" style would show me all the nodes in that style, and would be a very quick way to change the style to "in progress" or "finished" -- or I could delete the node because it's no longer relevant.

Being able to double-click a style to see the nodes of that style would be a big help to me. Maybe this would be useful to more than just me. 

Worth considering?

~ Bal

**Comment by macmarrum** [comment]
Please have a look at this script: https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/chooseStyleToFilterBy.groovy

**Reply by macmarrum** [reply]
The first run of the script creates a cloud with all styles, each as a child node. Select one of them and run the script again to filter your mindmap.

In the script there is a configuration section with true/false properties, which steer how the script behaves.

**Reply by wordmuse** [reply]
I really appreciate this. But I'm having trouble getting FP to recognize it as a script. I copied it into the FP scripts fold. Then I closed and reran FP. When I look at the scripts under the Tools menu, your script doesn't show up for me to select.

Thoughts?

~ Bal

**Reply by EdoFro** [reply]
maybe it’s looking into the FP v1.10 user directory or you put it in that folder and it’s looking into the FP v1.11 folder.
I had that problem once when changing FP version.

**Reply by abc16361** [reply]
I think it is on the menu. The ExecutionCodes sets it on Mac2.
<img width="305" alt="202305140643" src="https://github.com/freeplane/freeplane/assets/101362122/e27a17ed-bd02-4dc8-898d-98a543420df3">

**Reply by abc16361** [reply]
Just tested it, it's on the menu.
<img width="482" alt="202305141744" src="https://github.com/freeplane/freeplane/assets/101362122/a204003d-a743-4785-b76b-33eabbc5f1fa">
